addons/api_doc/controllers/api_doc.py

```diff
@@ -32,7 +33,7 @@ class DocController(http.Controller):
     documents.
     """
 
-    @http.route(['/doc', '/doc/<model_name>', '/doc/index.html'], type='http', auth='user')
+    @http.route(['/doc', '/doc/<model_name>', '/doc/index.html'], type='json2', auth='user')
```

Comment:
Keep this one http, we are rendering a html page, not sending away json stuff. The `return res` works by chance because we have an explicit case in the Json2Dispatcher `if isinstance(res, Response): return res`, but the `AccessError` would be serialized as json instead of as a pretty web page.